tests/test_ibsi_glcm.h

```diff
@@ -0,0 +1,196 @@
+#pragma once
+
+#include <gtest/gtest.h>
+
+#include "../src/nyx/roi_cache.h"
+#include "../src/nyx/parallel.h"
+#include "../src/nyx/features/glcm.h"
+#include "../src/nyx/features/pixel.h"
+#include "../src/nyx/environment.h"
+#include "test_data.h"
+#include "test_main_nyxus.h"
+
+#include <unordered_map> 
+
+// dig. phantom values for intensity based features
+static std::unordered_map<std::string, float> IBSI_glcm_values {
+    {"GLCM_DIFFERENCEAVERAGE", 1.42},
+    {"GLCM_DIFFERENCEVARIANCE", 2.9},
+    {"GLCM_DIFFERENCEENTROPY", 1.4},
+    {"GLCM_SUMAVERAGE", 4.28},
+    {"GLCM_SUMVARIANCE", 5.47},
+    {"GLCM_SUMENTROPY", 1.6}, 
+    {"GLCM_ANGULAR2NDMOMENT", 0.368},
+    {"GLCM_CONTRAST", 5.28},
+    {"GLCM_INVERSEDIFFERENCEMOMENT", 0.619},
+    {"GLCM_CORRELATION", -0.0121},
+    {"GLCM_INFOMEAS1", -0.155},
+    {"GLCM_INFOMEAS2", 0.487}
+
+};
+
+
+void test_glcm_feature(const AvailableFeatures& feature, const std::string& feature_name) {
+    double total = 0;
+    
+    LR roidata;
+    // Calculate features
+    GLCMFeature f;
+    Environment::skip_binning = true; 
+    GLCMFeature::angles = {0, 45, 90, 135};
+
+    // image 1
+    load_masked_test_roi_data (roidata, ibsi_phantom_z1_intensity, ibsi_phantom_z1_mask,  sizeof(ibsi_phantom_z1_mask) / sizeof(NyxusPixel));
+    ASSERT_NO_THROW(f.calculate(roidata));
+
+    // Initialize per-ROI feature value buffer with zeros
+    roidata.initialize_fvals();
+
+    // Retrieve values of the features implemented by class 'PixelIntensityFeatures' into ROI's feature buffer
+    f.save_value(roidata.fvals);
+ 
+
+    total += roidata.fvals[feature][0];
+    total += roidata.fvals[feature][1];
+    total += roidata.fvals[feature][2];
+    total += roidata.fvals[feature][3];
+
+    // image 2
+    // Calculate features
+    LR roidata1;
+    // Calculate features
+    GLCMFeature f1;
+    //GLCMFeature::n_levels = 6;
+    Environment::skip_binning = true; //<<< New!
```

Comment:
I suggest to also unit-test the 'skip_binning==false' scenario, just to make sure that GLCM (and other texture features - GLDM, GLRLM, etc) don't crash when we set skip_binning := false for production cases